tests/fundAccountPage.test.ts

```diff
@@ -0,0 +1,148 @@
+      "Friendbot: fund a testnet network account",
+    );
+  });
+
+  test("Shows futurenet network in the title if I change my network to futurenet", async ({
+    page,
+  }) => {
+    // Click network selector dropdown button
+    await page.getByTestId("networkSelector-button").click();
+    await expect(page.getByTestId("networkSelector-dropdown")).toBeVisible();
+
+    // Select Futurenet in the dropdown list
+    await page
+      .getByTestId("networkSelector-dropdown")
+      .getByTestId("networkSelector-option")
+      .filter({ has: page.getByText("Futurenet") })
+      .click();
+
+    // Network Submit button
+    const submitButton = page.getByTestId("networkSelector-submit-button");
+
+    // Select 'Futurenet' in the network dropdown list
+    await expect(submitButton).toHaveText("Switch to Futurenet");
+    await expect(submitButton).toBeEnabled();
+
+    // Click 'Switch to Futurenet' button
+    await submitButton.click();
+
+    await expect(page.getByTestId("networkSelector-button")).toHaveText(
+      "Futurenet",
+    );
+    await expect(page.locator("h1")).toHaveText(
+      "Friendbot: fund a futurenet network account",
+    );
+  });
+
+  test("By default, 'Public Key' input field is empty and buttons are disabled", async ({
+    page,
+  }) => {
+    await expect(page.locator("#generate-keypair-publickey")).toHaveValue("");
+
+    const getLumenButton = page
+      .getByTestId("fundAccount-buttons")
+      .getByText("Get lumens");
+    const fillInButton = page
+      .getByTestId("fundAccount-buttons")
+      .getByText("Fill in with generated key");
+
+    await expect(getLumenButton).toBeDisabled();
+    await expect(fillInButton).toBeDisabled();
+  });
+
+  test("Gets an error with an invalid public key in 'Public Key' field", async ({
+    page,
+  }) => {
+    const publicKeyInput = page.locator("#generate-keypair-publickey");
+
+    // Type in an invalid string in 'Public Key' input field
+    await publicKeyInput.fill("XLKDSFJLSKDJF");
+
+    await expect(publicKeyInput).toHaveAttribute("aria-invalid", "true");
+    await expect(
+      page.getByTestId("fundAccount-buttons").getByText("Get lumens"),
+    ).toBeDisabled();
+  });
+
+  test("Successfully funds an account when clicking 'Get lumens' with a valid public key", async ({
+    page,
+  }) => {
+    // Get a new public key
+    const publicKey =
+      "GDVOT2ALMUF3G54RBHNJUEV6LOAZCQQCARHEVNUPKGMVPWFC4PFN33QR";
+    const publicKeyInput = page.locator("#generate-keypair-publickey");
+    const getLumenButton = page
+      .getByTestId("fundAccount-buttons")
+      .getByText("Get lumens");
+
+    // Type in an invalid string in 'Public Key' input field
+    await publicKeyInput.fill(publicKey);
+
+    await expect(publicKeyInput).toHaveAttribute("aria-invalid", "false");
+    await expect(getLumenButton).toBeEnabled();
+
+    // Mock the friendbot api call
+    await page.route("**/*", async (route, request) => {
```

Comment:
mocking friendbot api call

Reply:
Nice! Could we use full route string here `*/**/?addr=GDVOT2ALMUF3G54RBHNJUEV6LOAZCQQCARHEVNUPKGMVPWFC4PFN33QR` and remove the `if` check?